packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -138,45 +139,45 @@ export class Snapshot {
     });
   }
 
-  public async waitUntilDone(
-    operationToWaitFor?: ResourceSnapshotsReportType | null,
-    options: WaitUntilDoneOptions = {},
-    iteratee = (_report: ResourceSnapshotsReportModel) => {}
+  public waitUntilDone(
+    operationToWaitFor: ResourceSnapshotsReportType | null,
+    options: Partial<WaitUntilDoneOptions> = {},
+    onRetryCb = (_report: ResourceSnapshotsReportModel) => {}
   ) {
+    const opts = {...Snapshot.defaultWaitOptions, options};
     const toMilliseconds = (seconds: number) => seconds * 1e3;
-    const opts = {
-      ...Snapshot.defaultWaitOptions,
-      ...options,
-    };
-    let timeout: NodeJS.Timeout;
-
-    await new Promise<void>((resolve, reject) => {
-      if (opts.wait > 0) {
-        timeout = setTimeout(() => {
-          const err = new SnapshotOperationTimeoutError(this);
-          clearInterval(interval);
-          reject(err);
-        }, toMilliseconds(opts.wait));
+
+    return retry(
+      this.waitUntilDoneRetryFunction(operationToWaitFor, onRetryCb),
+      {
+        retries: Infinity,
+        minTimeout: toMilliseconds(opts.waitInterval),
+        maxTimeout: toMilliseconds(opts.waitInterval),
```

Comment:
We want to retry every `opts.waitInterval` seconds.
Setting both limits of the Timeout ensure that this
`Math.min(random * minTimeout * Math.pow(factor, attempt), maxTimeout)` (see [node-retry](https://github.com/tim-kos/node-retry)) will always be eq to `opts.waitInterval`
given that: 
```
random=1 // i.e. no random
factor=2 // default value
attempt>=0 // 
```
 => Math.pow(factor,attempt)>=1
 => lower bound of the Math.min>=opts.waitInterval.
 => Math.min === opts.waitInterval